CHAID/column.py

```diff
@@ -123,15 +123,106 @@ def __getitem__(self, key):
 
     def __setitem__(self, key, value):
         self._arr[key] = value
-        return NominalColumn(np.array(self._arr), metadata=self.metadata, substitute=False)
+        return self
 
     def groups(self):
         return list(self._groupings.values())
 
     def possible_groupings(self):
-        return enumerate(combinations(self._groupings.keys(), 2))
+        return combinations(self._groupings.keys(), 2)
 
     def group(self, x, y):
         self._groupings[x] += self._groupings[y]
         del self._groupings[y]
         self._arr[self._arr == y] = x
+
+
+class OrdinalColumn(Column):
+    """
+    A column containing integer values that have an order
+    """
+    def __init__(self, arr=None, metadata=None,
+                 missing_id='<missing>', groupings=None, substitute=True):
+        super(self.__class__, self).__init__(arr, metadata, missing_id)
+
+        if substitute:
+            self._arr, self.orig_type = self.substitute_values(self._arr)
+
+        self._groupings = {}
+        if groupings is None:
+            for x in np.unique(self._arr):
+                self._groupings[x] = [x, x + 1, False]
+        else:
+            for x in np.unique(self._arr):
+                self._groupings[x] = list(groupings[x])
+        self._nan = np.array([np.nan]).astype(int)[0]
```

Comment:
@xulaus can you remember what this line is doing?

Reply:
Not exactly. I think we were using nan as a sentinel value that can combine with any item in the ordinal but we wanted the arrays to all be of the same dtype so we did this force cast.